### ISSUE TITLE
[Feature Request] WooCommerce 'Menu Cart' widget with customizable button text

### ISSUE THREAD

**Issue by DaCon** [elementor/elementor#5695]
## Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro. )

**What problem is your feature request going to solve? Please describe.**
I would like to change the button descriptions to the local language

![menu cart widget button descriptions](https://user-images.githubusercontent.com/1583919/45257824-bfe4f480-b3ac-11e8-891d-69e578ecd78d.png)

**Solution**
customizable button type and text for the menu buttons

**Comment by JoeDarko** [comment]
Is there any workaround to this? Where is the text loading from? Can it be translated by translation plugin? These strings are not even present in "Translating Wordpress" project. I'm building an ecommerce site for my client and i've just encountered this problem.

If it can't be translated then you've made this widget useless for every non english user. :/

**Comment by pr0pre** [comment]
https://github.com/pojome/elementor/issues/5495

**Comment by JoeDarko** [comment]
#6406

**Comment by pr0pre** [comment]
why  https://github.com/pojome/elementor/issues/5495  has been closed ?

**Comment by bainternet** [comment]
We are Going to change to WooCommerce translations instead of Elementor pro in the WooCommerce module.


[**@pr0pre**](https://github.com/pr0pre) commented on [Dec 13, 2018, 3:41 PM GMT+2](https://github.com/pojome/elementor/issues/5695#issuecomment-446971796 "2018-12-13T13:41:02Z - Replied by Github Reply Comments"):
> why  https://github.com/pojome/elementor/issues/5495  has been closed ?

because it's been merged with this.

Thanks.

**Comment by black-eye** [comment]
+1

**Comment by xhostAB** [comment]
+1 here too. Need to translate it to swedish

**Comment by addekk** [comment]
Hi Elementor Crew. Two weeks later there is still no solution? I need translate in Polish.

**Comment by jfbeaulieu420** [comment]
I was looking for the solution too and finally found the file where you can manually change the words. The file is this one : /wp-content/plugins/elementor-pro/modules/woocommerce/module.php

If you go to line 117, you will find this code : 
`<div class="elementor-menu-cart__subtotal">
			<strong><?php esc_attr_e( 'Sous-total', 'elementor-pro' ); ?>:</strong> <?php echo $sub_total; ?>
		</div>
		<div class="elementor-menu-cart__footer-buttons">
			<a href="<?php echo esc_url( wc_get_cart_url() ); ?>" class="elementor-button elementor-button--view-cart elementor-size-md">
				<span class="elementor-button-text"><?php esc_attr_e( 'Afficher le panier', 'elementor-pro' ); ?></span>
			</a>
			<a href="<?php echo esc_url( wc_get_checkout_url() ); ?>" class="elementor-button elementor-button--checkout elementor-size-md">
				<span class="elementor-button-text"><?php esc_attr_e( 'Payer', 'elementor-pro' ); ?></span>
			</a>
		</div>`

I already changed the words in french for me. You can just locate "Sous-total" - "Afficher le panier" - "Payer" and change it to your language. Worked for me, should work for you.

Hopefully, Elementor team will come with an easy way to change the language for this add-on.

Thanks

**Comment by MarciosilK** [comment]
> I was looking for the solution too and finally found the file where you can manually change the words. The file is this one : /wp-content/plugins/elementor-pro/modules/woocommerce/module.php
> 
> If you go to line 117, you will find this code :
> `<div class="elementor-menu-cart__subtotal"> <strong><?php esc_attr_e( 'Sous-total', 'elementor-pro' ); ?>:</strong> <?php echo $sub_total; ?> </div> <div class="elementor-menu-cart__footer-buttons"> <a href="<?php echo esc_url( wc_get_cart_url() ); ?>" class="elementor-button elementor-button--view-cart elementor-size-md"> <span class="elementor-button-text"><?php esc_attr_e( 'Afficher le panier', 'elementor-pro' ); ?></span> </a> <a href="<?php echo esc_url( wc_get_checkout_url() ); ?>" class="elementor-button elementor-button--checkout elementor-size-md"> <span class="elementor-button-text"><?php esc_attr_e( 'Payer', 'elementor-pro' ); ?></span> </a> </div>`
> 
> I already changed the words in french for me. You can just locate "Sous-total" - "Afficher le panier" - "Payer" and change it to your language. Worked for me, should work for you.
> 
> Hopefully, Elementor team will come with an easy way to change the language for this add-on.
> 
> Thanks

Deu certo!!!!!
Muito obrigado

**Comment by addekk** [comment]
Hi, maybe you did sth wrong. This is my code in functions.php file in Child theme:

`<?php
// Exit if accessed directly
if ( !defined( 'ABSPATH' ) ) exit;

// BEGIN ENQUEUE PARENT ACTION
// AUTO GENERATED - Do not modify or remove comment markers above or below:
         
if ( !function_exists( 'child_theme_configurator_css' ) ):
    function child_theme_configurator_css() {
        wp_enqueue_style( 'chld_thm_cfg_child', trailingslashit( get_stylesheet_directory_uri() ) . 'style.css', array( 'astra-theme-css','astra-menu-animation','astra-contact-form-7','woocommerce-layout','woocommerce-smallscreen','woocommerce-general' ) );
    }
endif;
add_action( 'wp_enqueue_scripts', 'child_theme_configurator_css', 20 );

// END ENQUEUE PARENT ACTION


/*Change text strings */
function my_text_strings( $translated_text, $text, $domain ) {
    switch ( $translated_text ) {
        case 'Subtotal' :
            $translated_text = __( 'Suma zamówienia', 'woocommerce' );
            break;
            case 'Checkout' :
            $translated_text = __( 'Finalizacja', 'woocommerce' );
            break;
            case 'Show Cart' :
            $translated_text = __( 'Pokaż koszyk', 'woocommerce' );
            break;
			case 'No products in the cart.' :
            $translated_text = __( 'Brak produktów w koszyku.', 'woocommerce' );
            break;
    }
    return $translated_text;
}
add_filter( 'gettext', 'my_text_strings', 20, 3 );`

**Comment by steffanhalv** [comment]
@bainternet  - We have a multilanguage site and stuck with this problem to. Any progress?

**Comment by shilo-ey** [comment]
[**@bainternet**](https://github.com/bainternet) commented on [Dec 13, 2018, 3:43 PM GMT+2](https://github.com/elementor/elementor/issues/5695#issuecomment-446972581 "2018-12-13T13:43:26Z - Replied by Github Reply Comments"):
> We are Going to change to WooCommerce translations instead of Elementor pro in the WooCommerce module.

This has been implemented in Elementor Pro v2.4.4

Thanks.

**Comment by gaspar-dega** [comment]
So I just ran into this problem as well. My site is in swedish and I just translated it to english. So my buttons are in swedish and I want to translate them to english. I have tried all these things here but nothing is working. Any ideas?

**Comment by weeb98** [comment]
I found
1. Install https://fa.wordpress.org/plugins/loco-translate/
2. Create your own language
3. Translate the word you want
4. Empty the cart
5. Product Add your desired word translated into the language of your site


My Cart Plugin ( Element Pack )
My problem was solved by translating into Persian

**Comment by mammadkoma** [comment]
1. Install https://fa.wordpress.org/plugins/loco-translate/
2. Go to wordpress dashboard
3. in right menu : loco translate > plugins
4. click on woocommerce
5. click on edit (bottom of your language)
6. search your word
7. select the finded word
8. update the text (bottom text area)
9. click on save

**Comment by Viktor-as** [comment]
Had the same problem. I was using TranslatePress to make my website multilingual. Tried everything here but nothing worked. So finally I found that I can change those translations in database, in wp_trp_gettext_lt_lt table (the table end depends on your language). 

But to see the changes I also had to enter my website in incognito mode.  Maybe this php code above works, just have to test it  in incognito mode.
